superpixel_classification/SuperpixelClassification/SuperpixelClassificationTensorflow.py

```diff
@@ -87,6 +96,19 @@ def predictLabelsForItemDetails(self, batchSize, ds, item, model, prog):
         catWeights = tf.nn.softmax(predictions)
         return catWeights, predictions
 
+    def findOptimalBatchSize(self, model, ds) -> int:
+        # Find an optimal batch_size
+        maximum_batchSize: int = 65536  # power of two
+        batchSize: int = 1
+        while batchSize < maximum_batchSize:
```

Comment:
When training the model, we might not have this many samples, so we should also stop when the batchsize is greater that the size of ds.

Reply:
Good, thank you.